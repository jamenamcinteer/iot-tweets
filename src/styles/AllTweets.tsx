import styled from "styled-components"
import { IPropsTheme } from "../interfaces/interfaces"

export const TweetsContainer = styled.div`
  background-color: ${(props: IPropsTheme) => props.theme.pageBackgroundColor};
  margin-top: 5rem;
  margin-bottom: -5rem;
  padding: 5rem 2rem;

  @media (min-width: 40rem) {
    padding: 5rem;
  }
`

export const TweetsGrid = styled.div`
  display: grid;
  grid-gap: 2rem;

  @media(min-width: 40rem) {
    grid-gap: 5rem;
    grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  }
`

export const TweetsHeader = styled.h2`
  font-size: ${(props: IPropsTheme) => props.theme.fontSizeLarge};
  font-weight: normal;
  margin: 0;
  padding-bottom: 3rem;
`

export const TweetUserProfileImageContainer = styled.div`
  border-radius: 9999px;
  box-shadow: ${(props: IPropsTheme) => props.theme.boxShadowSecondary};
  display: inline-block;
`

export const TweetUserProfileImage = styled.img`
  border-radius: 9999px;
  display: block;
`

export const TweetDate = styled.span`
  white-space: nowrap;
`

export const TweetMeta = styled.p`
  font-size: ${(props: IPropsTheme) => props.theme.fontSizeSmall};
`

export const TweetText = styled.p`
  font-size: ${(props: IPropsTheme) => props.theme.fontSizeMedium};
`