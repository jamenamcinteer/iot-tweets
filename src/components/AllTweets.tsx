import React from "react"
import moment from "moment"
import LazyLoad from "react-lazyload";
import { ITweets, IAllTweetsProps } from "../interfaces/interfaces"
import { TweetsContainer, TweetsGrid, TweetsHeader, TweetUserProfileImageContainer, TweetUserProfileImage, TweetDate, TweetMeta, TweetText } from "../styles/AllTweets"

function AllTweets({ twitterData }: IAllTweetsProps) {
  return (
    <TweetsContainer>
      <TweetsHeader>All 100 Tweets</TweetsHeader>
      <TweetsGrid>
      {twitterData.map((currentTweet: ITweets) => (
        <LazyLoad height={'100%'} key={currentTweet.id}>
          <div>
            <TweetUserProfileImageContainer>
              <TweetUserProfileImage src={currentTweet.user_profile_image_url_https} alt="" data-testid="tweet-img" />
            </TweetUserProfileImageContainer>
            <TweetMeta data-testid="tweet-meta">{currentTweet.user_screen_name} @{currentTweet.user_name} <TweetDate>Â· {moment(currentTweet.created_at, "ddd MMM D HH:mm:ss ZZ yyyy").fromNow()}</TweetDate></TweetMeta>
            <TweetText data-testid="tweet-text" dangerouslySetInnerHTML={currentTweet}></TweetText>
          </div>
        </LazyLoad>
      ))}
      </TweetsGrid>
    </TweetsContainer>
  )
}

export default AllTweets