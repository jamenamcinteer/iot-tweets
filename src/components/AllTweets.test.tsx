import React from 'react'
import { cleanup, render, fireEvent, wait } from "@testing-library/react"
import "jest-styled-components"
import { mockSuccessResponseTwitter } from "../__mocks__/mockTwitterApiResponse"
import AllTweets from './AllTweets'

afterEach(cleanup)

it('renders as expected', () => {
  const realDateNow = Date.now.bind(global.Date)
  const dateNowStub = jest.fn(() => 1574012671025)
  global.Date.now = dateNowStub

  const { asFragment } = render(<AllTweets twitterData={mockSuccessResponseTwitter} />)
  expect(asFragment()).toMatchSnapshot()

  global.Date.now = realDateNow
})