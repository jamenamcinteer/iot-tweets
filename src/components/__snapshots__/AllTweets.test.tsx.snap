// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 5rem;
  margin-bottom: -5rem;
  padding: 5rem 2rem;
}

.c2 {
  display: grid;
  grid-gap: 2rem;
}

.c1 {
  font-weight: normal;
  margin: 0;
  padding-bottom: 3rem;
}

@media (min-width:40rem) {
  .c0 {
    padding: 5rem;
  }
}

@media (min-width:40rem) {
  .c2 {
    grid-gap: 5rem;
    grid-template-columns: repeat(auto-fill,minmax(300px,1fr));
  }
}

<div
    class="c0"
  >
    <h2
      class="c1"
    >
      All 100 Tweets
    </h2>
    <div
      class="c2"
    >
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
      <div
        class="lazyload-placeholder"
        style="height: 100%;"
      />
    </div>
  </div>
</DocumentFragment>
`;
